File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
		int[] answer = new int[temperatures.length];

		// stack 선언
		Deque<Integer> stack = new ArrayDeque<>();
		for (int i = 0; i < temperatures.length; i++) {
			// 현재 온도가 스택에 있는 온도보다 높다면 꺼내서 결과 업데이트
			while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
				int last = stack.pop();
				answer[last] = i - last;
			}

			stack.push(i);
		}

		return answer;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.kt
import java.util.ArrayDeque

class Solution {
    fun dailyTemperatures(temperatures: IntArray): IntArray {
        val result = IntArray(temperatures.size)

        // stack 선언
        val stack: Deque<Int> = ArrayDeque()

        for (i in 0 until temperatures.size) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                var lastIndex = stack.pop()
                result[lastIndex] = i - lastIndex
            }

            stack.push(i)
        }
        return result
    }
}